Glyssen/Block.cs

```diff
@@ -913,6 +914,27 @@ public Block CombineWith(Block otherBlock)
 			return this;
 		}
 
+		/// <summary>
+		/// This is basically a simplified version of CombineWith which assumes we are just dealing with simple text
+		/// </summary>
+		public Block AppendTextFrom(Block blockWithTextToAppend)
+		{
+			var clone = Clone(true);
+
+			Debug.Assert(clone.BlockElements.Count == 1);
+			Debug.Assert(clone.BlockElements.OfType<ScriptText>().Count() == 1);
+			Debug.Assert(blockWithTextToAppend.BlockElements.Count == 1);
+			Debug.Assert(blockWithTextToAppend.BlockElements.OfType<ScriptText>().Count() == 1);
+
+			var baseBlockText = clone.GetText(false);
+			clone.BlockElements.OfType<ScriptText>().Single().Content = baseBlockText.CombineGuaranteeingWhitespaceBetween(blockWithTextToAppend.GetText(false));
+
+			if (clone.MatchesReferenceText && blockWithTextToAppend.MatchesReferenceText)
+				clone.ReferenceBlocks[0] = clone.ReferenceBlocks.Single().AppendTextFrom(blockWithTextToAppend.ReferenceBlocks.Single());
```

Comment:
Might never happen in real life, but if the clone does not have a matching reference text, but the blockWithTextToAppend does, wouldn't we probably want to clone the reference text block and stick it into clone.ReferenceBlocks?